src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -41,6 +42,25 @@ public class RandomAttribute : DataAttribute, IParameterDataSource
     {
         private RandomDataSource _source;
         private int _count;
+        private bool _distinct;
+
+        /// <summary>
+        /// If true, the random values will be all different
+        /// </summary>
+        public bool Distinct
+        {
+            get
+            {
+                return _distinct;
+            }
+            set
+            {
+                _distinct = value;
+
+                if (_source != null)
+                    _source.Distinct = _distinct;
```

Comment:
There's a constructor which only takes one param, `int count`, and source only gets created on GetData, and there are other constructors, with `int min` and `int max`, where the source is created, before the property Distinct gets set.